connect/runtime/src/main/java/org/apache/kafka/connect/storage/ConnectorOffsetBackingStore.java

```diff
@@ -279,8 +284,51 @@ public Future<Void> set(Map<ByteBuffer, ByteBuffer> values, Callback<Void> callb
             throw new IllegalStateException("At least one non-null offset store must be provided");
         }
 
+        boolean containsTombstones = values.containsValue(null);
+
+        // If there are tombstone offsets, then the failure to write to secondary store will
+        // not be ignored. Also, for tombstone records, we first write to secondary store and
+        // then to primary stores.
+        if (secondaryStore != null && containsTombstones) {
+            AtomicReference<Throwable> secondaryStoreTombstoneWriteError = new AtomicReference<>();
```

Comment:
It doesn't need to be, but it's cleaner that way. Otherwise we will need to make explicit null checks. This pattern is used at other places as well like [here](https://github.com/apache/kafka/blob/6368d14a1d8c37305290b8b89fb5990ad07aa4db/clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java#L430-L484)

Reply:
Aren't we still making explicit null checks here - https://github.com/apache/kafka/pull/13801/files#diff-0014cfa4c32bfde215cb10bee987e997d5182815fcf1a1245539375519f83f3dR325-R326? 🤔 

I'm fine with retaining it though, this isn't a blocking comment.

Reply:
haha yeah, that still exists. I am leaning on keeping it for now.

Reply:
Ok. I removed it.